conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureHandlerTest.java

```diff
@@ -80,6 +102,17 @@ public void after() {
         server.stop();
     }
 
+    @Test
+    public void invokesWrapperBeforeBlocking() {
+        when(wrapperObserver.control()).thenReturn(1);
+        execute();
+        //check that the first wrapper (the one that adds 1) is called before the one that adds 2.
+        assertThat(ImmutableList.copyOf(wrappersBeforeBlockingCallOrder).reverse())
```

Comment:
isn't this backwards?

Reply:
In what sense? Adding to the list add to the front so you have to reverse it to get the original handler

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/List.html#add-E-

> Appends the specified element to the end of this list

Reply:
I was so sure lists were doing append so that add would be O(1). Good to know!
Your fix is right then.